src/context/concrete_base.rs

```diff
@@ -210,25 +214,45 @@ impl ConcreteBaseAudioContext {
     /// The sample rate (in sample-frames per second) at which the `AudioContext` handles audio.
     #[allow(clippy::cast_precision_loss)]
     #[must_use]
-    pub fn sample_rate(&self) -> f32 {
+    pub(super) fn sample_rate(&self) -> f32 {
         self.inner.sample_rate.0 as f32
     }
 
     /// The raw sample rate of the `AudioContext` (which has more precision than the float
     /// [`sample_rate()`](BaseAudioContext::sample_rate) value).
     #[must_use]
-    pub fn sample_rate_raw(&self) -> SampleRate {
+    pub(super) fn sample_rate_raw(&self) -> SampleRate {
         self.inner.sample_rate
     }
 
+    /// This represents the number of seconds of processing latency incurred by
+    /// the `AudioContext` passing the audio from the `AudioDestinationNode`
+    /// to the audio subsystem.
+    // We don't do any buffering between rendering the audio and sending
+    // it to the audio subsystem, so this value is zero. (see Gecko)
+    #[allow(clippy::unused_self)]
+    #[must_use]
+    pub(super) const fn base_latency(&self) -> f64 {
+        0.
+    }
+
+    /// The estimation in seconds of audio output latency, i.e., the interval
+    /// between the time the UA requests the host system to play a buffer and
+    /// the time at which the first sample in the buffer is actually processed
+    /// by the audio output device.
+    #[must_use]
+    pub(super) fn output_latency(&self) -> f64 {
+        self.inner.output_latency.load()
```

Comment:
From the spec
> The [outputLatency](https://www.w3.org/TR/webaudio/#dom-audiocontext-outputlatency) attribute value does not change for the context’s lifetime as long as the connected audio output device remains the same.

But I do see it changing, are we just being more accurate than the spec mandates?

Reply:
I realize I generally use the editor's draft in which the spec seems to have evolved on this specific point 

https://webaudio.github.io/web-audio-api/#dom-audiocontext-outputlatency

> The [outputLatency](https://webaudio.github.io/web-audio-api/#dom-audiocontext-outputlatency) attribute value may change while the context is running or the associated audio output device changes. It is useful to query this value frequently when accurate synchronization is required.

So, we are ok here !

Reply:
Huh, interesting. Good to know